src/instructions/read.rs

```diff
@@ -1,76 +1,55 @@
 use super::instruction_id;
-use crate::endian::{read_u8_le, read_u16_le, read_u32_le, write_u16_le};
-use crate::{Bus, TransferError};
-
-pub struct ReadResponse<'a, ReadBuffer, WriteBuffer>
-where
-	ReadBuffer: AsRef<[u8]> + AsMut<[u8]>,
-	WriteBuffer: AsRef<[u8]> + AsMut<[u8]>,
-{
-	pub(crate) response: crate::Response<'a, ReadBuffer, WriteBuffer>,
-}
-
-impl<'a, ReadBuffer, WriteBuffer> ReadResponse<'a, ReadBuffer, WriteBuffer>
-where
-	ReadBuffer: AsRef<[u8]> + AsMut<[u8]>,
-	WriteBuffer: AsRef<[u8]> + AsMut<[u8]>,
-{
-	/// Get the ID of the motor.
-	pub fn motor_id(&self) -> u8 {
-		self.response.packet_id()
-	}
-
-	/// Get the read data as byte slice.
-	///
-	/// The individual registers of the motor are encoded as little-endian.
-	/// Refer to the online manual of your motor for the addresses and sizes of all registers.
-	pub fn data(&self) -> &[u8] {
-		self.response.parameters()
-	}
-}
+use crate::endian::write_u16_le;
+use crate::{bus::StatusPacket, Bus, Response, TransferError};
 
 impl<ReadBuffer, WriteBuffer> Bus<ReadBuffer, WriteBuffer>
 where
 	ReadBuffer: AsRef<[u8]> + AsMut<[u8]>,
 	WriteBuffer: AsRef<[u8]> + AsMut<[u8]>,
 {
-	/// Read an arbitrary number of bytes from a specific motor.
-	///
-	/// This function will not work correctly if the motor ID is set to [`packet_id::BROADCAST`][crate::instructions::packet_id::BROADCAST].
-	/// Use [`Self::sync_read`] to read from multiple motors with one command.
-	pub fn read(&mut self, motor_id: u8, address: u16, count: u16) -> Result<ReadResponse<ReadBuffer, WriteBuffer>, TransferError> {
+	/// Read an arbitrary number of bytes from multiple motors.
+	fn _read(&mut self, motor_id: u8, address: u16, count: u16) -> Result<StatusPacket<'_, ReadBuffer, WriteBuffer>, TransferError> {
```

Comment:
```suggestion
	fn read_raw(&mut self, motor_id: u8, address: u16, count: u16) -> Result<StatusPacket<'_, ReadBuffer, WriteBuffer>, TransferError> {
```

A leading underscore tells the compiler not to complain when the function is unused. We should avoid that.

We could use the name `read_raw()` to avoid that.